frontend/device/sdl/device.lua

```diff
@@ -42,34 +42,79 @@ function Device:init()
         self.input = require("device/input"):new{
             device = self,
             event_map = require("device/sdl/event_map_sdl2"),
-            handleMiscEv = function(device_input, ev)
-                -- bit of a hack for passing SDL events
+            handleSdlEv = function(device_input, ev)
+                local Geom = require("ui/geometry")
+                local TimeVal = require("ui/timeval")
+                local UIManager = require("ui/uimanager")
+
+                -- SDL events can remain cdata but are almost completely transparent
+                local SDL_MOUSEWHEEL = 1027
                 local SDL_DROPFILE = 4096
                 local SDL_WINDOWEVENT_RESIZED = 5
-                local w = 0
-                local h = 1
 
-                if ev.code == SDL_DROPFILE then
-                    local dropped_file_path = input.getDroppedFilePath()
+                if ev.code == SDL_MOUSEWHEEL then
+                    local scrolled_x = ev.value.x
+                    local scrolled_y = ev.value.y
+
+                    local up = 1
+                    local down = -1
+
+                    local pos = Geom:new{
+                        x = 0,
+                        y = 0,
+                        w = 0, h = 0,
+                    }
+
+                    local timev = TimeVal:new(ev.time)
+
+                    local fake_ges = {
+                        ges = "pan",
+                        distance = 200,
+                        relative = {
+                            x = 50*scrolled_x,
+                            y = 100*scrolled_y,
+                        },
+                        pos = pos,
+                        time = timev,
+                    }
+                    local fake_ges_release = {
+                        ges = "pan_release",
+                        distance = 200,
+                        relative = {
+                            x = 50*scrolled_x,
+                            y = 100*scrolled_y,
+                        },
+                        pos = pos,
+                        time = timev,
+                    }
+                    local fake_pan_ev = Event:new("Pan", nil, fake_ges)
+                    local fake_release_ev = Event:new("Gesture", fake_ges_release)
+
+                    if scrolled_y == down then
+                        fake_ges.direction = "north"
+                        UIManager:broadcastEvent(fake_pan_ev)
+                        UIManager:broadcastEvent(fake_release_ev)
+                    elseif scrolled_y == up then
+                        fake_ges.direction = "south"
+                        UIManager:broadcastEvent(fake_pan_ev)
+                        UIManager:broadcastEvent(fake_release_ev)
```

Comment:
I think these `UIManager:broadcastEvent(...)` should be `UIManager:sendEvent(...)`.
Otherwise, if I add sich support in a widget:
```lua
function PageBrowserWidget:onPan(arg, ges)
    if ges.mousewheel_direction then
        if ges.direction == "north" then
            self:onScrollRowDown()
        elseif ges.direction == "south" then
            self:onScrollRowUp()
        end
    end
    return true
end
```
to use the mousewheel there.... then, my Reader below also change pages :)

Thoughts?

Reply:
I think I actually considered that a feature when I wrote it, for example to be able to scroll with the dictionary search dialog up. But if something doesn't consume it that probably happens either way, both for better and for worse.

Or maybe I just accidentally used the wrong method. I'm not sure really. ;-)